### PR TITLE
add utxo_changes endpoint and update bindings

### LINKED ISSUE #305: API - get_milestone_utxo()
`milestones/:milestoneId/utxo-changes` endpoint has been added to HORNET nodes.

we need a corresponding `get_milestone_utxo()` in `iota.rs`.

### PR DESCRIPTION
**Pull request by Thoralf-M**

### CODE REVIEW

File: bindings/python/native/src/client/types.rs [hunk 1]
```diff
@@ -81,6 +84,13 @@ pub struct MilestoneDto {
     pub message_id: String,
 }
 
+#[derive(Debug, Clone, DeriveFromPyObject, DeriveIntoPyObject)]
+pub struct MilestoneUtxoChanges {
+    pub index: u32,
+    pub created_outputs: Vec<String>,
```
Comment (bernardoaraujor) [inline bindings/python/native/src/client/types.rs:90]: instead of `Vec<String>`, IMO we should have `Vec<UTXOInput>` in order to match `get_output()`

**Review by bernardoaraujor — CHANGES_REQUESTED** [review]

File: bindings/python/native/src/client/types.rs [hunk 2]
```diff
@@ -81,6 +84,13 @@ pub struct MilestoneDto {
     pub message_id: String,
 }
 
+#[derive(Debug, Clone, DeriveFromPyObject, DeriveIntoPyObject)]
+pub struct MilestoneUtxoChanges {
```
Comment (bernardoaraujor) [inline bindings/python/native/src/client/types.rs]: `MilestoneUTXOChanges`?

**Comment by bernardoaraujor** [comment]
@Thoralf-M just so you get an idea of what I'm aiming to achieve:
```
    // a bunch of stuff 
    //...
    let utxo_changes = match iota_client.get_milestone_utxo_changes(milestone_index).await {
        Ok(utxo_changes) => utxo_changes,
        Err(_) => return Err(ApiError::UnableToGetMilestoneUTXOChanges),
    };
    let transactions = vec![];
    for output_id in utxo_changes.created_outputs {
        let transaction_identifier = TransactionIdentifier {
            hash: output_id
        };
        transactions.push(Transaction {
            transaction_identifier: transaction_identifier,
            operations: xxxx
        });
    }
    let block = Block {
        block_identifier: block_identifier,
        parent_block_identifier: parent_block_identifier,
        timestamp: timestamp,
        transactions: transactions,
    };
    let response = BlockResponse { block };
    Ok(response)
```

File: bindings/python/native/src/client/types.rs (hunk 1 above)
Comment (Thoralf-M) [reply bindings/python/native/src/client/types.rs:90]: Yeah makes sense, but should be modified in Bee for all API endpoints, maybe in the future

**Review by bernardoaraujor — APPROVED** [review]

**Review by bingyanglin — APPROVED** [review]

**Pull request merged**